app/code/Magento/Theme/view/frontend/web/js/view/messages.js

```diff
@@ -8,32 +8,48 @@
  */
 define([
     'jquery',
+    'ko',
     'uiComponent',
     'Magento_Customer/js/customer-data',
     'underscore',
     'escaper',
     'jquery/jquery-storageapi'
-], function ($, Component, customerData, _, escaper) {
+], function ($, ko, Component, customerData, _, escaper) {
     'use strict';
 
     return Component.extend({
         defaults: {
             cookieMessages: [],
-            messages: [],
+            messages: ko.observableArray([]),
```

Comment:
This is a backward incompatible change. If you want to make it "observable", use `tracks` instead. More information you could find [here](https://devdocs.magento.com/guides/v2.4/ui_comp_guide/concepts/ui_comp_uielement_concept.html)

Reply:
I have added track variable in initObservable

Reply:
Current when digging here i found that magento not pass message via this `messages`. It's empty. It's never used
Magento get messages from `cookieStorage.get('mage-messages')`

Reply:
https://github.com/js-cookie/js-cookie Maybe this one will be use in near future